packages/compass-aggregations/src/components/stage-toolbar/option-menu.tsx

```diff
@@ -105,18 +47,36 @@ export const OptionMenu = ({
         );
       }}
     >
-      {menuItems.map((item) => (
-        <MenuItem
-          key={item.label}
-          data-text={item.label}
-          onClick={item.onClick}
-        >
-          <div className={menuItemStyles}>
-            <Icon color={palette.gray.dark2} glyph={item.icon} size="small" />
-            {item.label}
-          </div>
-        </MenuItem>
-      ))}
+      <OptionMenuItem
+        label="Add stage after"
+        icon="PlusWithCircle"
+        onClick={() => onAddStageClick(index)}
+        setMenuOpen={setMenuOpen}
```

Comment:
What benefit does it have to pass these two here instead of just `onClick` and then something like:

```:js
	onClick={() => {
        onAddStageClick(index);
        setMenuOpen(false);
    }}
```

Reply:
not a blocker though.

Reply:
This just ensures that `setMenuOpen(false)` is always called when `onClick` is called, not making that a choice of the caller.